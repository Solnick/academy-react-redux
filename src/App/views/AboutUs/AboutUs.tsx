import React from 'react';
import styles from './aboutUs.module.scss';
import View from '../../components/View/View';
import InfoBox from './InfoBox/InfoBox';
import infoBoxImg1 from '../../assets/images/infoBox1.png';
import infoBoxImg2 from '../../assets/images/infoBox2.png';
import infoBoxImg3 from '../../assets/images/infoBox3.png';

const aboutUsInfo = [
  {
    title: 'Lorem ipsum',
    text: 'Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.',
    img: infoBoxImg1,
  },
  {
    title: 'Lorem ipsum',
    text: 'Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.',
    img: infoBoxImg2,
  },
  {
    title: 'Lorem ipsum',
    text: 'Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.',
    img: infoBoxImg3,
  },
];

const AboutUs = () => (
  <View>
    <div className={styles.aboutUsWrapper}>
      {
        aboutUsInfo.map(({ text, title, img }) => <InfoBox key={img} text={text} title={title} img={img} />)
      }
    </div>
  </View>
);

export default AboutUs;
