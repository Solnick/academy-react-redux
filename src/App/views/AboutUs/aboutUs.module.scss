@import '../../style/variables';
@import '../../style/breakpoints';

.aboutUsWrapper {
  width: 100%;
  max-width: $view-width;
  background-color: #f8f8f8;
  display: flex;
  flex-direction: column;
  align-items: center;
  @media(min-width: 905px) {
    flex-direction: row;
    justify-content: space-between;
  }
}