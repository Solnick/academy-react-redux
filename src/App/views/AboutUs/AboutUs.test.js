import React from 'react';
import { shallow } from 'enzyme';
import AboutUs from './AboutUs';

describe('AboutUs component', () => {
  it('should render correctly', () => {
    const aboutUsInstance = shallow(<AboutUs />);
    expect(aboutUsInstance).toMatchSnapshot();
  });
});
