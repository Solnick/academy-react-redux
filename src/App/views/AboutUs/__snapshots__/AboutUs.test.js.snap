// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutUs />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="aboutUsWrapper"
      >
        <InfoBox
          img="infoBox1.png"
          text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
          title="Lorem ipsum"
        />
        <InfoBox
          img="infoBox2.png"
          text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
          title="Lorem ipsum"
        />
        <InfoBox
          img="infoBox3.png"
          text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
          title="Lorem ipsum"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InfoBox
            img="infoBox1.png"
            text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
            title="Lorem ipsum"
          />,
          <InfoBox
            img="infoBox2.png"
            text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
            title="Lorem ipsum"
          />,
          <InfoBox
            img="infoBox3.png"
            text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
            title="Lorem ipsum"
          />,
        ],
        "className": "aboutUsWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "infoBox1.png",
          "nodeType": "function",
          "props": Object {
            "img": "infoBox1.png",
            "text": "Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.",
            "title": "Lorem ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "infoBox2.png",
          "nodeType": "function",
          "props": Object {
            "img": "infoBox2.png",
            "text": "Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.",
            "title": "Lorem ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "infoBox3.png",
          "nodeType": "function",
          "props": Object {
            "img": "infoBox3.png",
            "text": "Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.",
            "title": "Lorem ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="aboutUsWrapper"
        >
          <InfoBox
            img="infoBox1.png"
            text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
            title="Lorem ipsum"
          />
          <InfoBox
            img="infoBox2.png"
            text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
            title="Lorem ipsum"
          />
          <InfoBox
            img="infoBox3.png"
            text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
            title="Lorem ipsum"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <InfoBox
              img="infoBox1.png"
              text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
              title="Lorem ipsum"
            />,
            <InfoBox
              img="infoBox2.png"
              text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
              title="Lorem ipsum"
            />,
            <InfoBox
              img="infoBox3.png"
              text="Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas."
              title="Lorem ipsum"
            />,
          ],
          "className": "aboutUsWrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "infoBox1.png",
            "nodeType": "function",
            "props": Object {
              "img": "infoBox1.png",
              "text": "Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.",
              "title": "Lorem ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "infoBox2.png",
            "nodeType": "function",
            "props": Object {
              "img": "infoBox2.png",
              "text": "Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.",
              "title": "Lorem ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "infoBox3.png",
            "nodeType": "function",
            "props": Object {
              "img": "infoBox3.png",
              "text": "Etiam ullamcorper. Suspendisse a pellentesque dui, non felis maecenas.",
              "title": "Lorem ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
